### PR TITLE
Allow multiple marks on a window

### LINKED ISSUE #2014: [enhancement] Allow for multiple marks on a single window
At the moment each window can only have one (unique) mark.  Marking an already marked window overwrites the old mark with the new one.  This is fine when you are using marks only to either manually tag a few windows or in one or two scripts/command chains. But if you want to make more extensive use of marks, for example mixing manual marks with scripts using them or having a larger number of scripts using them - things can go ugly. 

As only one mark can be placed on a window at a time, the chance that a previously set mark is inadvertently removed by another script/commands rises with the number of places where marks are used. This may break command combinations and could be especially annoying if manually set marks suddenly vanish.

I would like it to be possible to set more than one mark on a single window (similar to vim). The syntax of the currently available commands ``mark`` and ``unmark`` could remain the same for downward compatibility with older configurations. Although it would probably be useful if ``unmark`` were a bit more versatile than just remove one mark or remove all marks on all windows. For example by taking regex as argument and/or an option remove all marks on a specified window (focused or by criteria; e.g. ``unmark --window``). With active ``show_marks``, multiple marks could be shown as comma-separated list in the title bar.

### PR DESCRIPTION
**Pull request by Airblader**
fixes #2014 

Putting it on waiting for now because I still need to make `unmark` aware of matched windows.

### CODE REVIEW

**Comment by Airblader** [comment]
@stapelberg OK, you can review this now.

File: docs/userguide [hunk 1]
```diff
@@ -1671,7 +1671,7 @@ workspace::
 	the special value +__focused__+ to match all windows in the currently
 	focused workspace.
 con_mark::
-	Compares the mark set for this container, see <<vim_like_marks>>.
+	Compares the marks set for this container, see <<vim_like_marks>>.
```
Comment (stapelberg) [inline docs/userguide]: Please add that this matches when any of the container’s marks matches the specified mark.

File: docs/userguide [hunk 2]
```diff
@@ -2133,21 +2133,33 @@ The additional +--toggle+ option will remove the mark if the window already has
 this mark, add it if the window has none or replace the current mark if it has
 another mark.
 
+By default, a window can only have one mark. You can use the +--add+ flag to
+put more than one mark on a window. Note that you may have to use this flag in
+combination with +--toggle+ as well.
```
Comment (stapelberg) [inline docs/userguide]: The second sentence is not clear to me. Why do I have to use --toggle as well? Or do I have to use --add when I want to use --toggle? What are the circumstances? Can you add an example below?

**Comment by stapelberg** [comment]
Looks good overall.

**Comment by Airblader** [comment]
@stapelberg Done. I've also added a commit for `con_id=__focused__`.

**Pull request merged**